src/pixelator/pna/analysis/denoise.py

```diff
@@ -229,6 +232,36 @@ def denoise_one_core_layer(
     return nodes_to_remove
 
 
+def write_denoised_edgelist(
+    pxl: PNAPixelDataset, umis_to_remove: list, output_edgelist_path: str
+):
+    """Write a filtered edgelist (with specified UMIs removed) to a Parquet file.
+
+    This function takes an existing PNAPixelDataset, removes edges associated
+    with the specified UMIs, and writes the resulting edgelist to a Parquet file
+    at the specified path. It does not create a new pixel (.pxl) file or update
+    any AnnData object.
+
+    Args:
+        pxl (PNAPixelDataset): The original pixel dataset containing the edgelist.
+        umis_to_remove (list): A list of UMIs (nodes) to be removed from the edgelist.
+        output_edgelist_path (str): The file path where the filtered edgelist Parquet file will be saved.
+
+    """
+    with pxl.view as con:
+        con.execute(
+            f"""
+            COPY(
+                SELECT *
+                FROM edgelist
+                WHERE umi1 NOT IN (SELECT value FROM (SELECT UNNEST(?) AS value))
```

Comment:
Is the internal select here necessary? Or could it work to do `WHERE umi1 NOT IN UNNEST(?)`?